src/index.ts

```diff
@@ -27,30 +25,31 @@ export function _buildMetadataAccessor(type: string) {
       property = '/' + options.property;
     }
 
-    const reqOpts = extend(
-                        true, {
-                          uri: BASE_URL + '/' + type + property,
+    const reqOpts = Object.assign(
+                        {
+                          url: `${BASE_URL}/${type}${property}`,
                           headers: {'Metadata-Flavor': 'Google'}
                         },
-                        options) as _request.Options &
+                        options) as AxiosRequestConfig &
         {property?: string};
     delete reqOpts.property;
 
-    const retryRequestOpts = {noResponseRetries: 0};
-
-    return request(reqOpts, retryRequestOpts, (err, res, body) => {
-      if (callback) {  // for type safety; this should already always be true
-        if (err) {
-          callback(err);
-        } else if (!res) {
-          callback(new Error('Invalid response from metadata service'));
-        } else if (res.statusCode !== 200) {
-          callback(new Error('Unsuccessful response status code'), res);
-        } else {
-          callback(null, res, body);
-        }
-      }
-    });
+    axios(reqOpts)
+        .then(res => {
+          if (callback) callback(null, res, res.data);
```

Comment:
Based on the comment you removed; callback is always avialable; so you can do `callback!(...)` and drop the `if`.

Reply:
fixed